lib/synapse/service_watcher/zookeeper.rb

```diff
@@ -157,12 +157,32 @@ def discover
         end
       end
 
-      node = @zk.get(@discovery['path'], :watch => true)
-      begin
-        new_config_for_generator = parse_service_config(node.first)
-      rescue StandardError => e
-        log.error "synapse: invalid config data in ZK node at #{@discovery['path']}: #{e}"
-        new_config_for_generator = {}
+      # support for a separate 'generator_config_path' key, for reading the
+      # generator config block, that may be different from the 'path' key where
+      # we discover service instances. if generator_config_path is present and
+      # the value is "disabled", then skip all zk-based discovery of the
+      # generator config (and use the values from the local config.json
+      # instead).
+      if @discovery.has_key?('generator_config_path')
+        if @discovery['generator_config_path'] != 'disabled'
```

Comment:
using two conditionals won't work; if the key is present, but set to `'disabled'`, you won't fall back to the `else` condition below, and will end up with neither the config from zk nor the config from the config file. for clarity, i would declare a variable like so:

```ruby
generator_config_path = @discovery['generator_config_path']
if generator_config_path && generator_config_path != 'disabled'
  stuff
else
  other stuff
end
```
use_config_from_zk = @discovery.has_key?('genreator_confi

Reply:
That's the desired behavior:  If the key is present and set to "disabled", it should disable any zk based discovery, including the "path" block below.

Otherwise there is no way to disable the "path" lookup.